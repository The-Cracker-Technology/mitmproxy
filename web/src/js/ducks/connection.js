export const ConnectionState = {
    INIT: Symbol("init"),
    FETCHING: Symbol("fetching"), // WebSocket is established, but still fetching resources.
    ESTABLISHED: Symbol("established"),
    ERROR: Symbol("error"),
    OFFLINE: Symbol("offline"), // indicates that there is no live (websocket) backend.
}

const defaultState = {
    state: ConnectionState.INIT,
    message: null,
}

export default function reducer(state = defaultState, action) {
    switch (action.type) {

        case ConnectionState.ESTABLISHED:
        case ConnectionState.FETCHING:
        case ConnectionState.ERROR:
        case ConnectionState.OFFLINE:
            return {
                state: action.type,
                message: action.message
            }

        default:
            return state
    }
}

export function startFetching() {
    return { type: ConnectionState.FETCHING }
}

export function connectionEstablished() {
    return { type: ConnectionState.ESTABLISHED }
}

export function connectionError(message) {
    return { type: ConnectionState.ERROR, message }
}
export function setOffline() {
    return { type: ConnectionState.OFFLINE }
}
