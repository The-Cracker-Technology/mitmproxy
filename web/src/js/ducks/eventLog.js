import reduceStore from "./utils/store"
import * as storeActions from "./utils/store"

export const ADD               = 'EVENTS_ADD'
export const RECEIVE           = 'EVENTS_RECEIVE'
export const TOGGLE_VISIBILITY = 'EVENTS_TOGGLE_VISIBILITY'
export const TOGGLE_FILTER     = 'EVENTS_TOGGLE_FILTER'

const defaultState = {
    visible: false,
    filters: { debug: false, info: true, web: true, warn: true, error: true },
    ...reduceStore(undefined, {}),
}

export default function reduce(state = defaultState, action) {
    switch (action.type) {

        case TOGGLE_VISIBILITY:
            return {
                ...state,
                visible: !state.visible
            }

        case TOGGLE_FILTER:
            const filters = { ...state.filters, [action.filter]: !state.filters[action.filter] }
            return {
                ...state,
                filters,
                ...reduceStore(state, storeActions.setFilter(log => filters[log.level]))
            }

        case ADD:
        case RECEIVE:
            return {
                ...state,
                ...reduceStore(state, storeActions[action.cmd](action.data, log => state.filters[log.level]))
            }

        default:
            return state
    }
}

export function toggleFilter(filter) {
    return { type: TOGGLE_FILTER, filter }
}

export function toggleVisibility() {
    return { type: TOGGLE_VISIBILITY }
}

export function add(message, level = 'web') {
    let data = {
        id: Math.random().toString(),
        message,
        level,
    }
    return {
        type: ADD,
        cmd: "add",
        data
    }
}
