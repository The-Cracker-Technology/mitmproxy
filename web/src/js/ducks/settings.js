import { fetchApi } from '../utils'

export const RECEIVE        = 'SETTINGS_RECEIVE'
export const UPDATE         = 'SETTINGS_UPDATE'
export const REQUEST_UPDATE = 'REQUEST_UPDATE'

const defaultState = {

}

export default function reducer(state = defaultState, action) {
    switch (action.type) {

        case RECEIVE:
            return action.data

        case UPDATE:
            return {
                ...state,
                ...action.data,
            }

        default:
            return state
    }
}

export function update(settings) {
    fetchApi.put('/settings', settings)
    return { type: REQUEST_UPDATE }
}
