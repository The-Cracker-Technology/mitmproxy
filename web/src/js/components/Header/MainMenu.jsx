import React, { Component } from "react"
import PropTypes from 'prop-types'
import { connect } from "react-redux"
import FilterInput from "./FilterInput"
import { update as updateSettings } from "../../ducks/settings"
import { setFilter, setHighlight } from "../../ducks/flows"

MainMenu.title = "Start"

export default function MainMenu() {
    return (
        <div className="menu-main">
            <FlowFilterInput/>
            <HighlightInput/>
            <InterceptInput/>
        </div>
    )
}

export function setIntercept(intercept) {
    updateSettings({ intercept })
}

const InterceptInput = connect(
    state => ({
        value: state.settings.intercept || '',
        placeholder: 'Intercept',
        type: 'pause',
        color: 'hsl(208, 56%, 53%)'
    }),
    { onChange: setIntercept }
)(FilterInput);

const FlowFilterInput = connect(
    state => ({
        value: state.flows.filter || '',
        placeholder: 'Search',
        type: 'search',
        color: 'black'
    }),
    { onChange: setFilter }
)(FilterInput);

const HighlightInput = connect(
    state => ({
        value: state.flows.highlight || '',
        placeholder: 'Highlight',
        type: 'tag',
        color: 'hsl(48, 100%, 50%)'
    }),
    { onChange: setHighlight }
)(FilterInput);
