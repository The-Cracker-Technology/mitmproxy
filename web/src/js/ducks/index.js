import { combineReducers } from "redux"
import eventLog from "./eventLog"
import flows from "./flows"
import settings from "./settings"
import ui from "./ui/index"
import connection from "./connection"
import options from './options'

export default combineReducers({
    eventLog,
    flows,
    settings,
    connection,
    ui,
    options,
})
