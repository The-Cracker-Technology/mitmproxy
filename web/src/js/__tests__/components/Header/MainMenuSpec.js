jest.mock('../../../ducks/settings')

import React from 'react'
import renderer from 'react-test-renderer'
import  MainMenu, { setIntercept } from '../../../components/Header/MainMenu'
import { Provider } from 'react-redux'
import { update as updateSettings } from '../../../ducks/settings'
import { TStore } from '../../ducks/tutils'

describe('MainMenu Component', () => {
    let store = TStore()

    it('should render and connect to state', () => {
        let provider = renderer.create(
            <Provider store={store}>
                <MainMenu/>
            </Provider>),
            tree = provider.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should handle change on interceptInput', () => {
        setIntercept('foo')
        expect(updateSettings).toBeCalledWith({ intercept: 'foo' })
    })
})
