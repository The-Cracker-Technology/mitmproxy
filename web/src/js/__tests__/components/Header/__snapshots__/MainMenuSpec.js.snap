// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu Component should render and connect to state 1`] = `
<div
  className="menu-main"
>
  <div
    className="filter-input input-group"
  >
    <span
      className="input-group-addon"
    >
      <i
        className="fa fa-fw fa-search"
        style={
          Object {
            "color": "black",
          }
        }
      />
    </span>
    <input
      className="form-control"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Search"
      type="text"
      value="~u foo"
    />
  </div>
  <div
    className="filter-input input-group"
  >
    <span
      className="input-group-addon"
    >
      <i
        className="fa fa-fw fa-tag"
        style={
          Object {
            "color": "hsl(48, 100%, 50%)",
          }
        }
      />
    </span>
    <input
      className="form-control"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Highlight"
      type="text"
      value="~a bar"
    />
  </div>
  <div
    className="filter-input input-group"
  >
    <span
      className="input-group-addon"
    >
      <i
        className="fa fa-fw fa-pause"
        style={
          Object {
            "color": "hsl(208, 56%, 53%)",
          }
        }
      />
    </span>
    <input
      className="form-control"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Intercept"
      type="text"
      value=""
    />
  </div>
</div>
`;
